website/docs/runtime.md

```diff
@@ -95,28 +94,190 @@ type annotations:
+example at Stripe, we use `.on_failure` to attach team ownership information to
+a failure.
 
-<!-- TODO(jez) Document .on_failure / .checked once API is stable. -->
+The `T::Configuration` handler we wrote branches on whether `.on_failure` was
+provided, which means the logging behavior is opt in. This is nice because it
+means the default behavior is still to make problems with types fail loudly and
+early, rather than silently as a log. If we wanted, we could have inverted this:
 
-It's possible to change the runtime behavior by defining callbacks. See
-[Runtime Configuration](tconfiguration.md) for more about what callbacks are
-available and how to register them.
+```ruby
+T::Configuration.call_validation_error_handler = lambda do |signature, opts|
+  #                                                     ┌────┐
+  if signature.on_failure && signature.on_failure[0] == :raise
+    raise TypeError.new(opts[:pretty_message])
+  else
+    puts opts[:pretty_message]
+  end
+end
+
+# ...
+
+#                                       ┌─────────────────┐
+sig {params(argv: T::Array[String]).void.on_failure(:raise)}
+```
+
+With this `T::Configuration` handler, the default is to log, and we can use
+`.on_failure` to opt specific sigs into raising on failure.
+
+For more on the various `T::Configuration` handlers, see
+[Runtime Configuration](tconfiguration.md).
+
+## `.checked`: Whether to check in the first place
+
+> **Careful!** Opting out of runtime checks can significantly degrade the
+> trustworthiness of type signatures. Only disable the runtime after
+> understanding the tradeoffs. See [Gradual Type Checking](gradual.md) to learn
+> more.
+
+In our examples above with `.on_failure`, every method call had the runtime type
+checks run to determine whether the `T::Configuration` handler should be called
+in the first place. This comes with an overhead, and we carefully audit and
+monitor the performance of `sorbet-runtime` as a result. The overhead of these
+checks are usually very small.
+
+But in some cases, especially when calling certain methods in tight loops or
+other latency-sensitive paths, the overhead of even doing the checks (regardless
+of what happens on failure) is prohibitively expensive. To handle these cases,
+Sorbet offers `.checked(...)` which declares in what environments a sig should
+be checked:
+
+```ruby
+# (1) Runtime checks always run.
+sig {params(xs: T::Array[String]).void.checked(:always)}
+
+# (2) Runtime checks only run in "tests" (see below).
+# In non-tests, this sig specifically has no runtime overhead.
+sig {params(xs: T::Array[String]).void.checked(:tests)}
+
+# (3) Never runs the runtime checks. Careful!
+sig {params(xs: T::Array[String]).void.checked(:never)}
+```
+
+If `.checked(...)` is omitted on a sig, the default is `.checked(:always)`. The
+default checked level can also be configured. For example:
+
+```ruby
+T::Configuration.default_checked_level = :tests
```

Comment:
I think it would be helpful to mention that this setting cannot be changed later on in the code and it must be set right after Sorbet is loaded.